<launch>
	<arg name="input" default="csi://0"/>
	<arg name="input_width" default="640"/>
	<arg name="input_height" default="480"/>
	<arg name="input_codec" default="unknown"/>
	<arg name="framerate" default="30"/>

	<arg name="lin_gain" default="60"/>
	<arg name="ang_gain" default="40"/>
	<arg name="max_spd" default="60"/>

	<param name="robot_description" command="cat $(find sandwich)/urdf/sandwich.urdf" />

	<node pkg="ros_deep_learning" type="video_source" name="video_source" output="screen">
		<param name="resource" value="$(arg input)"/>
		<param name="width" value="$(arg input_width)"/>
		<param name="height" value="$(arg input_height)"/>
		<param name="loop" value="0"/>
		<param name="framerate" value="$(arg framerate)"/>
		<remap from="/video_source/raw" to="/camera/image_raw"/>
	</node>
	
	<node pkg="sandwich" type="motcontrol.py" name="motcontrol" output="screen">
		<param name="lin_gain" value="$(arg lin_gain)"/>
		<param name="ang_gain" value="$(arg ang_gain)"/>
		<param name="max spd" value="$(arg max_spd)"/>
	</node>

	<include file="$(find sandwich_remote)/launch/bacon.launch"/>

	<node pkg="sandwich" type="camera_info_pub.py" name="camerainfo" output="screen"/>
	<node pkg="image_proc" type="image_proc" name="image_proc" ns="camera" output="screen"/>
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

</launch>
